File: src/main/java/gg/uhc/migration/configuration/PositiveIntegerReader.java
package gg.uhc.migration.configuration;

import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.configuration.InvalidConfigurationException;

public class PositiveIntegerReader extends IntegerReader {

    protected final boolean allowZero;

    public PositiveIntegerReader(boolean allowZero) {
        this.allowZero = allowZero;
    }

    @Override
    protected Integer read(ConfigurationSection section, String key) throws InvalidConfigurationException {
        int integer = super.read(section, key);

        if (allowZero ? integer < 0 : integer <= 0) {
            throw up(WRONG_TYPE, key, allowZero ? ">= 0" : "> 0", integer);
        }

        return integer;
    }
}

File: src/main/java/gg/uhc/migration/configuration/PositiveIntegerReader.kt
package gg.uhc.migration.configuration

import org.bukkit.configuration.ConfigurationSection

fun ConfigurationSection.toPositiveInteger(key: String, allowZero: Boolean): Int {
    val read = this.toInteger(key)

    if (read < 0)
        throw wrongType(key, if (allowZero) ">= 0" else "> 0", read)

    if (read > 0)
        return read

    if (!allowZero)
        throw wrongType(key, "> 0", 0)

    return read
}